index.js

```diff
@@ -100,3 +100,37 @@ function proxySignals (child) {
     }
   }
 }
+
+/**
+ * Starts forwarding standard IO (stdin, stdout, stderr) to `child` through `parent`.
+ *
+ * @param parent Parent process.
+ * @param child Child Process.
+ * @return `unproxy` function to stop the forwarding.
+ * @internal
+ */
+function proxyStdio(parent, child) {
+  if (typeof child.stdin === 'object' && child.stdin !== null) {
+    parent.stdin.pipe(child.stdin)
+  }
+  if (typeof child.stdout === 'object' && child.stdout !== null) {
+    child.stdout.pipe(parent.stdout)
+  }
+  if (typeof child.stderr === 'object' && child.stderr !== null) {
+    child.stderr.pipe(parent.stderr)
+  }
+
+  return unproxyStdio;
```

Comment:
Since `proxyStdio` and `unproxyStdio` has almost similar code, I'd suggest this rewrite:

```javascript
function proxyFactory(proxy) {
  var pu = proxy ? 'pipe' : 'unpipe';
  
  return function () {
    if (typeof child.stdin === 'object' && child.stdin !== null) {
      parent.stdin[pu](child.stdin)
    }
    if (typeof child.stdout === 'object' && child.stdout !== null) {
      child.stdout[pu](parent.stdout)
    }
    if (typeof child.stderr === 'object' && child.stderr !== null) {
      child.stderr[pu](parent.stderr)
    }
  }
}

var proxyStdio = proxyFactory(true);
var unproxyStdio = proxyFactory(false);

```

I don't know, just ideas... 🤷‍♂️

Reply:
I pondered about it but ultimately chose to go with the simplest solution (two functions) because it only duplicates a few lines. It also gives some symmetry to all the proxy functions.